models/DataObject/Traits/ContextPersistenceTrait.php

```diff
@@ -25,6 +25,65 @@
  */
 trait ContextPersistenceTrait
 {
+    protected function prepareMyCurrentRelations(
+        Localizedfield|\Pimcore\Model\DataObject\Fieldcollection\Data\AbstractData|\Pimcore\Model\DataObject\Objectbrick\Data\AbstractData|Concrete $object,
+        array $params
+    ): array {
+        if ($object instanceof Concrete) {
+            $relations = $object->retrieveRelationData(['fieldname' => $this->getName(), 'ownertype' => 'object']);
+        } elseif ($object instanceof AbstractData) {
+            $relations = $object->getObject()->retrieveRelationData(
+                [
+                    'fieldname' => $this->getName(),
+                    'ownertype' => 'fieldcollection',
+                    'ownername' => $object->getFieldname(),
+                    'position' => $object->getIndex(),
```

Comment:
https://github.com/pimcore/pimcore/blob/291f133bc562a3985b216876197c2f4bb6601c91/models/DataObject/Fieldcollection/Data/AbstractData.php#L44-L47 wouldn't match and be flagged as existing entry because `position `is `===` to strings